gapis/resolve/dependencygraph2/graph_visualization/graph_structure_test.go

```diff
@@ -0,0 +1,284 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graph_visualization
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+)
+
+func getSortedKeys(input map[int]int) []int {
+	sortedKeys := []int{}
+	for key := range input {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Ints(sortedKeys)
+	return sortedKeys
+}
+
+func getSortedKeysForNodes(input map[int]*node) []int {
+	sortedKeys := []int{}
+	for key := range input {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Ints(sortedKeys)
+	return sortedKeys
+}
+
+func addNodeByIdAndLabel(g *graph, id int, label string) {
+	newNode := g.getNewNode(id, label)
+	g.addNode(newNode)
+}
+
+func addNodeByIdAndLabelAndNameAndAttributes(g *graph, id int, label, name, attributes string) {
+	newNode := g.getNewNode(id, label)
+	newNode.name = name
+	newNode.attributes = attributes
+	g.addNode(newNode)
+}
+
+func areEqualGraphs(t *testing.T, wantedGraph *graph, obtainedGraph *graph) bool {
+
+	if wantedGraph.getNumberOfNodes() != obtainedGraph.getNumberOfNodes() {
+		t.Errorf("The numbers of nodes are different %v != %v\n", wantedGraph.getNumberOfNodes(), obtainedGraph.getNumberOfNodes())
+		return false
+	}
+	if wantedGraph.getNumberOfEdges() != obtainedGraph.getNumberOfEdges() {
+		t.Errorf("The numbers of edges are different %v != %v\n", wantedGraph.getNumberOfEdges(), obtainedGraph.getNumberOfEdges())
+		return false
+	}
+
+	wantedSortedIdNodes := getSortedKeysForNodes(wantedGraph.nodeIdToNode)
+	obtainedSortedIdNodes := getSortedKeysForNodes(obtainedGraph.nodeIdToNode)
+	if !reflect.DeepEqual(wantedSortedIdNodes, obtainedSortedIdNodes) {
+		t.Errorf("The nodes ID are different in the graphs\n")
+		return false
+	}
+
+	for _, id := range wantedSortedIdNodes {
+		wantedNode := wantedGraph.nodeIdToNode[id]
+		obtainedNode := obtainedGraph.nodeIdToNode[id]
+		if !reflect.DeepEqual(wantedNode.label, obtainedNode.label) {
+			t.Errorf("The labels from nodes with ID %d are different %v != %v\n", id, wantedNode.label, obtainedNode.label)
+			return false
+		}
+		if !reflect.DeepEqual(wantedNode.name, obtainedNode.name) {
+			t.Errorf("The name from nodes with ID %d are different %v != %v\n", id, wantedNode.name, obtainedNode.name)
+			return false
+		}
+		wantedInNeighbourIdSorted := getSortedKeys(wantedNode.inNeighbourIdToEdgeId)
+		obtainedInNeighbourIdSorted := getSortedKeys(obtainedNode.inNeighbourIdToEdgeId)
+		if !reflect.DeepEqual(wantedInNeighbourIdSorted, obtainedInNeighbourIdSorted) {
+			t.Errorf("The in-Neighbours ID are different for nodes with ID %d\n", id)
+			return false
+		}
+		wantedOutNeighbourIdSorted := getSortedKeys(wantedNode.outNeighbourIdToEdgeId)
+		obtainedOutNeighbourIdSorted := getSortedKeys(obtainedNode.outNeighbourIdToEdgeId)
+		if !reflect.DeepEqual(wantedOutNeighbourIdSorted, obtainedOutNeighbourIdSorted) {
+			t.Errorf("The out-Neighbours ID are different for nodes with ID %d\n", id)
+			return false
+		}
```

Comment:
Yes , this is intentional. Because the edge IDs are given in order starting from 1, then functions such as removeNodeById could not be possible to test, because the edges IDs are different even if the set of edges of two graphs are the same. This is also why currently I am not using reflect.DeepEqual(wantedNode.inNeighbourIdToEdgeId, obtainedNode.inNeighbourIdToEdgeId) directly, because the values (edge IDs) could not be the same, even if the graphs consist of the same set of edges.